app/src/sharedTest/java/org/oppia/android/app/policies/PoliciesActivityTest.kt

```diff
@@ -0,0 +1,201 @@
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.AssetModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.networking.NetworkConnectionDebugUtilModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [PoliciesActivity]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = PoliciesActivityTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class PoliciesActivityTest {
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<PoliciesActivity> = ActivityTestRule(
+    PoliciesActivity::class.java, /* initialTouchMode= */
+    true, /* launchActivity= */
+    false
+  )
+
+  @Inject
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testPoliciesActivity_hasCorrectPrivacyPolicyActivityLabel() {
+    activityTestRule.launchActivity(createPoliciesActivity(PolicyPage.PRIVACY_POLICY))
+    val titleToolbar =
+      activityTestRule.activity.findViewById<Toolbar>(R.id.policies_activity_toolbar)
+
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    assertThat(titleToolbar.title).isEqualTo(context.getString(R.string.privacy_policy_title))
+  }
+
+  @Test
+  fun testPoliciesActivity_hasCorrectTermsOfServiceActivityLabel() {
+    activityTestRule.launchActivity(createPoliciesActivity(PolicyPage.TERMS_OF_SERVICE))
+    val titleToolbar =
+      activityTestRule.activity.findViewById<Toolbar>(R.id.policies_activity_toolbar)
+
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    assertThat(titleToolbar.title).isEqualTo(context.getString(R.string.terms_of_service_title))
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  private fun createPoliciesActivity(policyPage: PolicyPage): Intent {
```

Comment:
Prefer not having control flow like this. Instead, just create the exact activity needed within the test--that should make the test clearer and more specific.

Reply:
Done.